internal/app/machined/pkg/runtime/v1alpha1/bootloader/adv/syslinux/syslinux.go

```diff
@@ -116,13 +119,17 @@ func (a ADV) SetTagBytes(t uint8, val []byte) (ok bool) {
 			continue
 		}
 
+		// overflow check
+		if i+2+len(val) > AdvSize-4-2 {
+			return false
+		}
+
 		length := uint8(len(val))
 
 		a[i] = t
-
 		a[i+1] = length
 
-		copy(a[i+2:uint8(i+2)+length], val)
```

Comment:
this is the core problem: `uint8(257) == 1`, so this wraps around and builds wrong second index